trackmania.js

```diff
@@ -504,29 +526,54 @@ export class TrackmaniaFacade {
         return res;
     }
 
-    /**
-     * 
-     * @param {LiveService} live_service 
-     * @param {JSON} track_info 
-     * @param {Number} [length=10]
-     * @param {Boolean} [onlyWorld=true]
-     * @param {Number} [offset=0]
-     * @returns {Promise<JSON>}
-     */
-    async getLeaderboardInfo(track_info, length = 25, onlyWorld = true, offset = 0) {
-        let lb_info = {
-            positions: [],
-            accountIds: [],
-            times: [],
+    async generateLeaderboardField(recordsJSON) {
+        const { name, records } = recordsJSON;
+        let field = {
+            name: name,
+            value: `\`\`\`RANK | TIME      | PLAYER\n-----+-----------+-----------------`,
+            inline: false,
+        };
+        let accountIds = [];
+        let players = [];
+        if (records.length > 0) {
+            records.forEach(record => accountIds.push(record.accountId));
+            const accounts = await this.oauthService.fetchAccountNames(accountIds);
```

Comment:
This whole function `generateLeaderboardField` is great, we should have more functions that just handle constructing "view" information (stuff going back to Discord) from the API responses we've received from Trackmania APIs. These would allow snapshot testing or just faster testing in Discord if you plugged in some dummy data from Trackmania.

On that note, is it possible to move this `fetchAccountNames` call outside of this function, and possibly moving this `generateLeaderboardField` method to a new class / different file?